content/02/02.05/script.js.adoc

```diff
@@ -1,3 +1,187 @@
 include::./title.adoc[]
 
-TODO
+shot::[1]
+Once we have the boilerplate in place, we can start working on the actual code for our scenario.
+
+.steps.js
+[source,js]
+----
+const { Given, When, Then } = require('cucumber');
+
+Given('Lucy is located {int}m from Sean', function (distance) {
+  console.log(distance);
+  return 'pending';
+});
+
+When('Sean shouts “free bagels at Sean’s”', function () {
+  // Write code here that turns the phrase above into concrete actions
+  return 'pending';
+});
+
+Then('Lucy hears Sean’s message', function () {
+  // Write code here that turns the phrase above into concrete actions
+  return 'pending';
+});
+----
+
+shot::[2]
+To implement the first step, we need to create a couple of person objects, with the specified distance between them. We could write it like this:
+
+shot::[3]
+We should also remove the `pending` status, since we are now defining our step.
+
+.steps.js
+[source,js]
+----
+const { Given, When, Then } = require('cucumber');
+
+Given('Lucy is located {int}m from Sean', function (distance) {
+  this.lucy = new Person
+  this.sean = new Person
+  this.lucy.move_to(distance)
+});
+
+When('Sean shouts “free bagels at Sean’s”', function () {
+  // Write code here that turns the phrase above into concrete actions
+  return 'pending';
+});
+
+Then('Lucy hears Sean’s message', function () {
+  // Write code here that turns the phrase above into concrete actions
+  return 'pending';
+});
+----
+
+Now we have two instances of person, one representing Lucy, and one representing Sean.
+
```

Comment:
```suggestion
video::02.05.animation.mp4[]
```


To keep things simple, we’re going to assume all people are situated on a line: a one-dimensional co-ordinate system. We can always introduce proper geo-locations later. We’ll place Sean in the centre, and Lucy 15 metres away from Sean.

This might not be the design we’ll end up with once this is all working, but it’s a decent place to start.